lessons/cryptography/Encoding/encoding.md

```diff
@@ -0,0 +1,118 @@
+# [Cryptography] Encoding
+*Learn about encoding, a way to reformat information.*
+
+![banner alt text](.rsrc/banner.png)
+
+# Objectives
+- ASCII
+- Unicode
+- Base64
+
+# Introduction
+Encoding refers to converting information into a specific form. Although encoding and encryption may sound similar, these two terms should not be confused. Encryption refers to converting information into a different form in order to keep information secret, such that decryption would be difficult without special knowledge. Encoding also converts information, but makes no attempt to prevent other people from obtaining the original message. In fact, encoding and decoding are simply opposite processes. Encoding is primarily used to ensure efficient transmission or storage of information. 
+
+The most common types of encoding include ASCII, unicode, and base64. A tool called [CyberChef](https://gchq.github.io/CyberChef/) can be useful for encoding and decoding messages. 
+
+1. ASCII
+2. Unicode 
+3. Base64
+
+# Lesson
+## ASCII
+ASCII stands for American Standard Code for Information Interchange, and is a character encoding standard. Characters that can be encoded include uppercase letters, lowercase letters, digits, and many punctuation marks. The standard ASCII table includes 128 encoded characters using a 7-bit binary code. There is also an extended ASCII table (the modern standard) that includes 256 characters, which is the max number of characters that can be encoded with an 8-bit binary code. The table below shows both the basic and extended ASCII tables. 
+
+![Standard and Extended ASCII (https://i.pcmag.com/imagery/encyclopedia-terms/7-bit-ascii-_aschart.fit_lim.size_640x.gif)](.rsrc/7-bit-and-8-bit-ascii.png)
+
+In the table above, the numbers 0 through 127 represent all of the standard ASCII characters, and the numbers 128 through 255 represent all of the extended ASCII characters. The numbers 0 through 255 are displayed in decimal form, which can then be converted to binary or hexadecimal form. 
+
+For example, if we want to encode the message **"Hi"** in ASCII in binary form, we would first look up the decimal representation of the uppdercase letter **"H"**, which is **72**, and the decimal representation of the lowercase letter **"i"**, which is **105**. We would then convert **72** and **105** to binary, which would get us the ASCII encoded message of **"1001000 1101001"**. 
```

Comment:
We're introducing a lot of new concepts here but I think this is pretty good for now. Curious how well people will retain this information.